IHP/Job/Dashboard.hs

```diff
@@ -0,0 +1,530 @@
+    -- Can be defined as any arbitrary view.
+    makeDetailView :: (?context :: ControllerContext, ?modelContext :: ModelContext) => job -> IO SomeView
+    makeDetailView job = do
+        pure $ SomeView $ GenericShowView job
+
+    -- | The content of the page that will be displayed for the "new job" form of this job.
+    -- By default, only the submit button is rendered. For additonal form data, define your own implementation.
+    -- See 'GenericNewJobView' for a guide on how it can look.look
+    -- Can be defined as any arbitrary view, but it should be a form.
+    makeNewJobView :: (?context :: ControllerContext, ?modelContext :: ModelContext) => IO SomeView
+    makeNewJobView = pure $ SomeView $ GenericNewJobView @job
+
+    -- | The action run to create and insert a new value of this job into the database.
+    -- By default, create an empty record and insert it.
+    -- To add more data, define your own implementation.
+    createNewJob :: (?context :: ControllerContext, ?modelContext :: ModelContext) => IO ()
+    createNewJob = do
+        newRecord @job |> create
+        pure ()
+
+
+
+data TableView a = TableView [a]
+
+-- A TableView is only a View if it's contents are TableViewable.
+instance forall a. (TableViewable a) => View (TableView a) where
+    html (TableView rows) = renderTable rows
+
+-- | Creates a form with a single small button that links to the Create Job form for some 'DisplayableJob;.
+newJobFormForTableHeader :: forall job. (DisplayableJob job) => Html
+newJobFormForTableHeader =
+        let
+            t = tableName @job
+            link :: Text = "/jobs/CreateJob?tableName=" <> t
+        in [hsx|
+            <form action={link}>
+                <button type="submit" class="btn btn-primary btn-sm">+ New Job</button>
+            </form>
+        |]
+
+-- | Defines a set of necessary functions to display some data 'a' in a table format given a list of '[a]'.
+class TableViewable a where
+    tableTitle :: Text
+
+    -- | Headers that describe each column in the table
+    tableHeaders :: [Text]
+
+    -- | Optional HTML that will be placed in the top right of the table. Used for a "create new" button.
+    createNewForm :: Html
+    createNewForm = [hsx||]
+
+    renderTableRow :: a -> Html
+
+    renderTable :: [a] -> Html
+    renderTable rows =
+        let
+            title = tableTitle @a
+            headers = tableHeaders @a
+            renderRow = renderTableRow @a
+            newForm = createNewForm @a
+        in [hsx|
+        <div>
+            <div class="d-flex justify-content-between align-items-center">
+                <h3>Job type: {title}</h3>
+                {newForm}
+            </div>
+            <table class="table table-sm">
+                <thead>
+                    <tr>
+                        {forEach headers renderHeader}
+                    </tr>
+                </thead>
```

Comment:
Could we also have a retry button inside the table?

![image](https://user-images.githubusercontent.com/2072185/112450643-e09bac80-8d54-11eb-86b6-aad5b9d8d2f1.png)

This could save one click :)